### PR TITLE
use incluster tekton hub Api endpoint in pipeline builder

### PR DESCRIPTION
**Pull request by karthikjeeyar**
**Fixes**: 

https://issues.redhat.com/browse/ODC-6136

**Analysis / Root cause**: 

Pipeline builder only supports public tektonhub endpoint and as part of this ticket we need to support incluster tekton hub.

**Solution Description**: 

Read the tektonHub (hub) CR to get the API, UI urls and use them to populate the tasks in pipeline builder. If the In cluster tekton hub is not installed then fallback to public endpoint.

**Screen shots / Gifs for design review**: 

API call is made to in-cluster tektonhub url 
 ie. `https://tekton-hub-api-karthik.apps.13apr22.devcluster.openshift.com/v1/resources` 

Screenshot 2022-04-13 at 8 09 56 PM

**Unit test coverage report**: 

```
    ✓ should return custom path url if the baseurl param is passed as a second argument (1ms)
    
 useInClusterTektonHubURLs:
    ✓ should return public tekton hub endpoint incase of hub CR not found (11ms)
    ✓ should return public tekton hub endpoint when hub is installed and if status field missing (2ms)
    ✓ should return public tekton hub endpoint when hub is installed but urls are missing (1ms)
    ✓ should return incluster tekton hub endpoint when hub is installed (1ms)
    ✓ should return the apiUrl and uiUrl when the api call the hub has urls in the status (1ms)
 ```

**Test setup:**

1. Install OSP 1.7.0 from custom catalogsource -> https://artifacts.ospqa.com/builds/1.7.0
2. Follow the tektonhub docs to install incluster tektonhub 
    https://github.com/tektoncd/operator/blob/main/docs/TektonHub.md

**Browser conformance**: 

- [x] Chrome
- [x] Firefox
- [x] Safari
- [ ] Edge

@siamaksade @serenamarie125

### CODE REVIEW

**Review by rottencandy — COMMENTED** [review]
Looks like i18n files need updating.

File: frontend/packages/pipelines-plugin/src/components/catalog/apis/tektonHub.ts [hunk 1]
```diff
@@ -42,29 +45,47 @@ export type TektonHubTask = {
...
 export const TEKTON_HUB_ENDPOINT = `https://hub.tekton.dev`;
 export const TEKTON_HUB_INTEGRATION_KEY = 'enable-devconsole-integration';
 
-export const getHubUIPath = (path: string = ''): string => {
+export const getHubUIPath = (path: string = '', baseURL: string = TEKTON_HUB_ENDPOINT): string => {
   if (!path) {
     return null;
   }
-  return path ? `${TEKTON_HUB_ENDPOINT}/${path}` : TEKTON_HUB_ENDPOINT;
+  return path ? `${baseURL}/${path}` : baseURL;
```
[the remark this answers is not in the document]
Comment (karthikjeeyar) [reply frontend/packages/pipelines-plugin/src/components/catalog/apis/tektonHub.ts]: @rottencandy Yes, you are right. We dont need another condition below. Udpated it.

**Review by rottencandy — APPROVED** [review]
/lgtm
Tested on cluster, working as expected.
/label qe-approved

**Comment by karthikjeeyar** [comment]
Propagating docs/px Labels from [epic](https://issues.redhat.com/browse/ODC-6236)
/label docs-approved
/label px-approved

**Pull request merged**